File: Seccion 17 - Archivos o Ficheros en C++/Seccion_17_Dart/09_leer_binario.dart

import 'dart:io';
import 'dart:convert';

class Registro {

  String nombre;
  String apellidos;

  Registro();

}

void escribir() {

  final File file = new File("prueba.dat");

  Registro persona = new Registro();

  stdout.writeln("\tPidiendo datos");
  stdout.write("Digite su nombre: ");
  persona.nombre = stdin.readLineSync();
  stdout.write("Digite sus apellidos: ");
  persona.apellidos = stdin.readLineSync();

  List<int> bytes = utf8.encode("${persona.nombre} ${persona.apellidos}");

  file.writeAsBytesSync(bytes);
}

void leer() async {

  final File file = new File("prueba.dat");

  Registro persona = new Registro();

  List<int> bytes = await file.readAsBytes();
  String datos = utf8.decode(bytes);

  stdout.writeln("\tMostrando datos.");
  persona.nombre = datos.split(" ")[0];
  persona.apellidos = datos.split(" ")[1];
  print("Nombre: ${persona.nombre}\nApellidos: ${persona.apellidos}");

}
void main() {

  escribir();
  leer();

}

File: Seccion 17 - Archivos o Ficheros en C++/Seccion_17_Python/09_leer_binario.py

class Registro:
    
    def __init__(self, nombre='', apellidos=''):
        self.nombre = nombre
        self.apellidos = apellidos
    
    def __str__(self):
        return f"{self.nombre} {self.apellidos}"

def escribir():
    
    archivoB = open("prueba.dat", "wb")
    
    persona = Registro()
    
    print("\tPidiendo datos")
    persona.nombre = input("Digite un nombre: ")
    persona.apellidos = input("Digite un apellido: ")
    
    archivoB.write(str(persona).encode())
    archivoB.close()

def leer():
    
    archivoB = open("prueba.dat", "rb")
    
    persona = Registro()
    
    print("\n\tMostrando datos")
    datos = archivoB.read().decode().split()
    persona.nombre = datos[0]
    persona.apellidos = datos[1]
    print("Nombre: {}\nApellidos: {}".format(persona.nombre, persona.apellidos))

def main():
    
    escribir()
    leer()

if __name__ == "__main__":
    main()